test/idle-source-e2e/idle_source_test.go

```diff
@@ -115,26 +112,27 @@ func (is *IdleSourceSuite) TestIdleKeyedReducePipelineWithKafkaSource() {
 
 	done := make(chan struct{})
 	go func() {
-		startTime := time.Now().Add(-10000 * time.Hour)
+		startTime := time.UnixMilli(60000)
 		for i := 0; true; i++ {
 			select {
 			case <-ctx.Done():
 				return
 			case <-done:
 				return
 			default:
-				// send message to both partition for first 1000 messages for overcome the kafka source lazy loading wm publisher.
+				// send message to both partition for first 600 messages for overcome the kafka source lazy loading wm publisher.
 				// after that send message to only one partition. so that idle source will be detected and wm will be progressed.
 				SendMessage(topic, "data", generateMsg("1", startTime), 0)
-				if i < 2000 {
+				if i < 600 {
 					SendMessage(topic, "data", generateMsg("2", startTime), 1)
 				}
+				time.Sleep(100 * time.Millisecond)
```

Comment:
isn't this sleep a lot for 600 messages?

Reply:
We give about 1 minute for Kafka partition lazy loading. It might not take that long, but sometimes creating a Kafka consumer can take a while.